go/kbnm/hostmanifest/whitelist.go

```diff
@@ -0,0 +1,61 @@
+package hostmanifest
+
+import (
+	"encoding/json"
+	"os"
+	"os/user"
+	"path/filepath"
+)
+
+// whitelistPath is used for installing the whitelist as a JSON into a given
+// path on disk.
+type whitelistPath struct {
+	// Root is the path of the machine-global NativeMessage whitelists.
+	Root string
+	// Home is the path of the NativeMessage whitelists for regular users.
+	Home string
+	// Overlay is an optional path to prefix all paths with. Useful for
+	// generating whitelists for Linux distribution packaging.
+	Overlay string
+}
+
+func (w *whitelistPath) path(u *user.User, appID string) string {
+	if u.Uid == "0" {
+		return filepath.Join(w.Overlay, w.Root, appID+".json")
+	}
```

Comment:
Ok moved the overlay stuff around in this commit, should better support what we need now: https://github.com/keybase/client/pull/7484/commits/1f70e04f9a1c97b94805d5382c019c307ff80d06

```
$ KBNM_INSTALL_ROOT=1 KBNM_INSTALL_OVERLAY=$PWD/overlay ./kbnm install
$ tree overlay/
overlay/
└── Library
    ├── Application Support
    │   ├── Chromium
    │   │   └── NativeMessagingHosts
    │   │       └── io.keybase.kbnm.json
    │   └── Mozilla
    │       └── NativeMessagingHosts
    │           └── io.keybase.kbnm.json
    └── Google
        └── Chrome
            └── NativeMessagingHosts
                └── io.keybase.kbnm.json
```